torch/csrc/jit/export.cpp

```diff
@@ -899,6 +848,140 @@ void dump(const onnx::ModelProto& model, std::ostream& stream, size_t indent) {
+          url_prefix, model_def.GetDescriptor()->file()->pool()));
+    ::google::protobuf::util::Status convert_result =
+      ::google::protobuf::util::BinaryToJsonString(resolver.get(),
+          url_prefix + "/" + model_def.GetDescriptor()->full_name(),
+          model_def.SerializeAsString(), &output);
+    if (!convert_result.ok()) {
+      std::stringstream ss;
+      ss << convert_result;
+      AT_ERROR(ss.str());
+    }
+    auto record_id = writer_.writeRecord(output.data(), output.size());
+    writer_.writeEndOfFile();
+
+  }
+
+ private:
+  void convertToModel(const script::Module& module,
+      torch::ModelDef* model_def) {
+    model_def->set_name("script-model");
+    model_def->set_producer_name("pytorch");
+    model_def->set_producer_version("1.0"); // TODO: set the producer version using appropriate function call
+    model_def->set_proto_version(torch::ProtoVersion::PROTO_VERSION_NEWEST);
+    std::string main_module_name = "";
+    collectParamsInfo(module, main_module_name);
+    convertModule(module, main_module_name, model_def->mutable_main_module());
+  }
+  void collectParamsInfo(const script::Module& module, const std::string& prefix) {
+    for (const auto& elem: module.get_parameters()) {
+      const script::NamedParameter& param = elem.value();
+      parameterMap_[param.slot()] = prefix + param.name;
+    }
+    for (const auto& elem: module.get_modules()) {
+      collectParamsInfo(*elem->module, prefix + elem.key() + ".");
+    }
+  }
+  void convertModule(const script::Module& module, const std::string& name,
+      torch::ModuleDef* module_def) {
+    module_def->set_name(name);
+    for (const auto& elem: module.get_parameters()) {
+      torch::ParameterDef* param_def = module_def->add_parameters();
+      convertParameter(elem.value(), param_def);
+    }
+    for (auto& elem: module.get_methods()) {
+      torch::MethodDef* method_def = module_def->add_methods();
+      convertMethod(*elem.value(), method_def);
+    }
+    for (const auto& elem: module.get_modules()) {
+      torch::ModuleDef* sub_def = module_def->add_submodules();
+      convertModule(*elem->module, elem.key(), sub_def);
+    }
+  }
+  void convertParameter(const script::NamedParameter& param,
+      torch::ParameterDef* param_def) {
+    param_def->set_name(param.name);
+    param_def->set_is_buffer(param.is_buffer);
+    param_def->set_require_gradient(param.slot()->requires_grad());
+    convertTensor(*(param.slot()), param_def->mutable_tensor());
+  }
+  void convertTensor(const at::Tensor& tensor,
+      caffe2::TensorProto* tensor_proto) {
+    for (auto d : tensor.sizes()) {
+      tensor_proto->add_dims(d);
+    }
+    tensor_proto->set_data_type(
+        caffe2::TypeMetaToDataType(at::scalarTypeToTypeMeta(
+            tensor.type().scalarType())));
+    tensor_proto->set_storage_type(caffe2::TensorProto_StorageType_EXTERNAL);
+    caffe2::ExternalDataProto* external_data = tensor_proto->mutable_external_data();
```

Comment:
why are strides external data?

Reply:
Right now, the strides will be only used together with external data. I will move it back to TensorProto in the following PR.

Reply:
So right now TensorProto also support putting the data inline. If the data is inline - there's not need to put strides as the storage is by definition not shared.